### DISCUSSION TITLE
esmurl

### DISCUSSION THREAD

**Discussion by jcbhmr** [tinylibs/rfcs#4] [Show and tell]
https://github.com/jcbhmr/esmurl
👆 I made that and I'm "shopping" for a larger organization to transfer it to. I came across @unjs; I also want to open a discussion with them. This @tinylibs organization might be another place for such a package. 🤷‍♂️ Hence the discussion!

This is what it lets you do:

```js
// https://cdn.com/lib.js         👇 This import.meta is serialized into the self-contained URL
const workerURL = esmurl(import.meta, async () => {
  //                                👇 This is properly resolved to https://cdn.com/more.js even
  //                                      though it's being stringified and put into a blob: URL
  const moreLibCode = await import("./more.js")
  onmessage = e => postMessage(moreLibCode.compute(e.data))
})
const worker = new Worker(workerURL, { type: "module" })
```
And it even works with bundlers! It does so by regex-magicing all the `import()` into `__import__()` and then at runtime resolving any `__import__("./more.js")` via the user-provided import.meta object's `importMeta.url`

Yes, it can be renamed to some tiny* name if you want. Don't worry about the users; there are none yet. 😜

Why do I want to foist this onto an organization? If I am on vacation or something, somebody else can still do stuff! ...and I think it's a cool way to immediately boost eyes on such an infant project. 😅

**Comment by Aslemammad** [comment]
Another Neat idea by Jacob, You're welcome to have this in tinylibs as tinyesmurl or something like that. let me know (+ the name) so I create a repo for it so you bring it.
